files/image_config/config-setup/config-setup

```diff
@@ -99,28 +103,123 @@ run_hookdir() {
     return $exit_status
 }
 
+initalize_configdb()
+{
+    # Create correct argument based on per-asic name space in multi-npu platforms
+    if [[ ! -z "$2" ]]; then
+        NAMESPACE_ARGUMENT=${NAMESPACE_OPTION}${NAMESPACE_PREFIX}$2
+        IP_NETNS_CMD=${IP_NETNS_CMD_PREFIX}${NAMESPACE_PREFIX}$2
+    else
+        NAMESPACE_ARGUMENT=$2
+        IP_NETNS_CMD=$2
+    fi 
+    
+    if [ "$1" == "init" ]; then
+        # case when config db need to be created/initialized from minigraph
+        sonic-db-cli ${NAMESPACE_ARGUMENT} CONFIG_DB FLUSHDB
+        sonic-cfggen -H -m -j ${INIT_CFG_JSON} ${NAMESPACE_ARGUMENT} --write-to-db
+        sonic-db-cli ${NAMESPACE_ARGUMENT} CONFIG_DB SET "CONFIG_DB_INITIALIZED" "1"
+        sonic-cfggen -d --print-data > ${CONFIG_DB_PATH}${CONFIG_DB_PREFIX}$2${CONFIG_DB_SUFFIX}
+        if [[ ($NUM_ASIC -gt 1) && ( -z "$2") ]]; then
+            return 0
+        fi
+        # this will be run only in per-asic namespace only in multi-npu platforms
+        ${IP_NETNS_CMD} config qos reload
+        ${IP_NETNS_CMD} pfcwd start_default
+    elif [ "$1" == "reload" ]; then
+        # case when config db is already present from previous image file system
+        sonic-cfggen ${NAMESPACE_ARGUMENT} -j ${CONFIG_DB_PATH}${CONFIG_DB_PREFIX}$2${CONFIG_DB_SUFFIX} --write-to-db
+    fi
+
+    return 0
+}
+configdb_migrator()
+{
+    if [[ ! -z "$2" ]]; then
+        NAMESPACE_ARGUMENT=${NAMESPACE_OPTION}${NAMESPACE_PREFIX}$2
+    else
+        NAMESPACE_ARGUMENT=$2
```

Comment:
suggest explicitly assign NAMESPACE_ARGUMENT to empty string.

Reply:
@yxieca Done